Engine/Engine.cs

```diff
@@ -245,11 +245,15 @@ public void Run(AlgorithmNodePacket job, string assemblyPath)
                         // Algorithm runtime error:
                         if (algorithm.RunTimeError != null)
                         {
-                            throw algorithm.RunTimeError;
+                            throw new Exception("Algorithm.RunTimeError", algorithm.RunTimeError); //Keep original exception as inner exception to preserve stack trace
                         }
                     }
                     catch (Exception err)
                     {
+                        while (err != null && err.InnerException != null)
+                        {
+                            err = err.InnerException; //Find inner most exception as this will be most useful to algorithm designer
```

Comment:
Understand your point.
Could we make the below change though to prevent us from losing stack trace from the inner exception?
While re-throwing the exception it seems to lose the original stack trace unless u throw a new exception with the original exception as an inner exception.
`throw algorithm.RunTimeError;"`
to
`throw new Exception("Algorithm.RunTimeError", algorithm.RunTimeError)`